plugins/woocommerce/tests/php/src/Internal/Admin/Settings/PaymentProviders/WooPayments/WooPaymentsServiceTest.php

```diff
@@ -0,0 +1,4903 @@
+				array(
+					WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED => time(),
+				),
+				array(
+					'is_store_connected'  => false,
+					'has_connected_owner' => false,
+				),
+				array(
+					'has_account'       => false,
+					'has_valid_account' => false,
+					'test_account'      => false,
+				),
+			),
+			'payment_methods - stored started and completed' => array(
+				WooPaymentsService::ONBOARDING_STEP_PAYMENT_METHODS,
+				WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED,
+				array(
+					WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED => time() - 10,
+					WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED => time(),
+				),
+				array(
+					'is_store_connected'  => false,
+					'has_connected_owner' => false,
+				),
+				array(
+					'has_account'       => false,
+					'has_valid_account' => false,
+					'test_account'      => false,
+				),
+			),
+			'wpcom_connection - clean slate'      => array(
+				WooPaymentsService::ONBOARDING_STEP_WPCOM_CONNECTION,
+				WooPaymentsService::ONBOARDING_STEP_STATUS_NOT_STARTED,
+				array(),
+				array(
+					'is_store_connected'  => false,
+					'has_connected_owner' => false,
+				),
+				array(
+					'has_account'       => false,
+					'has_valid_account' => false,
+					'test_account'      => false,
+				),
+			),
+			'wpcom_connection - nothing stored with working connection' => array(
+				WooPaymentsService::ONBOARDING_STEP_WPCOM_CONNECTION,
+				WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED,
+				array(),
+				array(
+					'is_store_connected'  => true,
+					'has_connected_owner' => true,
+				),
+				array(
+					'has_account'       => false,
+					'has_valid_account' => false,
+					'test_account'      => false,
+				),
+			),
+			'wpcom_connection - nothing stored with partial connection' => array(
+				WooPaymentsService::ONBOARDING_STEP_WPCOM_CONNECTION,
+				WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED,
+				array(),
+				array(
+					'is_store_connected'  => true,
+					'has_connected_owner' => false,
+				),
+				array(
+					'has_account'       => false,
+					'has_valid_account' => false,
+					'test_account'      => false,
+				),
+			),
+			'wpcom_connection - stored started with no connection data' => array(
+				WooPaymentsService::ONBOARDING_STEP_WPCOM_CONNECTION,
+				WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED,
+				array(
+					WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED => time() - 10,
+				),
+				array(
+					'is_store_connected'  => false,
+					'has_connected_owner' => false,
+				),
+				array(
+					'has_account'       => false,
+					'has_valid_account' => false,
+					'test_account'      => false,
+				),
+			),
+			'wpcom_connection - stored started with working connection' => array(
+				WooPaymentsService::ONBOARDING_STEP_WPCOM_CONNECTION,
+				WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED,
+				array(
+					WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED => time() - 10,
```

Comment:
could it happen in reality?

Reply:
Of course. Start, then bail and set up the connection through other means (like the Jetpack plugin or the WooPayments client, or WooShipping).